src/components/data-pane/dataset-selector.tsx

```diff
@@ -29,6 +29,10 @@ export class DatasetSelector extends React.PureComponent<DatasetSelectorProps, {
     this.onDatasetChange = this.onDatasetChange.bind(this);
   }
 
+  public componentDidUpdate() {
```

Comment:
Not sure why we need to trigger this action here?  
I'm afraid we are going to trigger the query more often than we need to.